chainer/training/triggers/interval.py

```diff
@@ -40,9 +39,11 @@ def __call__(self, trainer):
         """
         updater = trainer.updater
         if self.unit == 'epoch':
-            prev = self.count
-            self.count = updater.epoch_detail // self.period
-            return prev != self.count
+            epoch = updater.epoch_detail
+            previous_epoch = updater.previous_epoch_detail
```

Comment:
`getattr` is safer in the terms of not raising any exception.
But setting `previous_epoch(_detail)` to `None` causes undesired behavior (same as #2244).
I think it is better to raise exception (or warning) when `updater` does not have `previous_epoch_detail` property.